File: Lista1/src/exec5.java
/**
 *
 * @author 11800910
 */
public class exec5 {

    public static void main(String[] args) {
        // cabeçalho
        System.out.println("EXERCICIO 5: Primeiro primo menor\n");

        // validação da entrada num (análogo ao feito em aula)
        int num = 0;
        boolean leu = false;
        while(!leu){
            leu = true;
            System.out.printf("Digite um inteiro: ");
            try {
                num = EntradaTeclado.leInt();
                if( num < 0){
                    System.out.println("Número deve ser positivo!!");
                    leu = false;
                }
            } catch (Exception e) {
            System.out.println("O valor que vc digitou não é um número inteiro");
                leu = false;
            }
        }
        // com num validado, vamos procurar o proximo menor num
        boolean primo = true;
        int menor_primo = num;
        int i;
        // loop indo de num a 0 procurando um primo
        while(menor_primo > 0){
            // teste de numero primo
            primo = true;
            for(i = 2; i < menor_primo; i++){
                if (menor_primo % i == 0){
                    primo = false;
                    break;
                }
            }
            // achou o primeiro numero primo
            if(primo == true) break;
            menor_primo--;
        }
        // resposta
        System.out.printf("O primeiro menor primo depois de %d é %d\n", num, menor_primo);

    }
}

File: Lista2/exec5.py
'''
 *
 * @author 11800910
'''

def valida_entrada():
    while(True):
        try:
            num = int(input('Digite um inteiro: '))
            if num < 0:
                print('Número deve ser positivo!!')
                continue
            break
        except ValueError:
            print('O valor que vc digitou não é um número inteiro')
    return num

if __name__ == '__main__':

    # cabeçalho
    print('EXERCICIO 5: Primeiro primo menor\n')
    
    # valdação da entrada 
    num = valida_entrada()

    # com num validado, vamos procurar o proximo menor num
    primo = True
    menor_primo = num

    # loop indo de num a 0 procurando um primo
    while menor_primo > 0:
        # teste de numero primo
        primo = True
        for i in range(2, menor_primo):
            if (menor_primo % i) == 0:
                primo = False
                break
        if primo == True:
            break
        menor_primo -= 1

    # resposta
    print(f'O primeiro menor primo depois de {num} é {menor_primo}')